header-rewrite-filter/header_processor.cc

```diff
@@ -190,13 +190,15 @@ namespace HeaderRewriteFilter {
         return absl::OkStatus();
     }
 
-    absl::Status SetBoolProcessor::executeOperation() {
+    std::pair<absl::Status, bool> SetBoolProcessor::executeOperation(bool negate) {
+        bool result;
         try {
-            result_ = matcher_(source_);
+            result = matcher_(source_);
         } catch (std::exception& e) {
-            return absl::InvalidArgumentError("failed to perform boolean match");
+            return std::make_pair(absl::InvalidArgumentError("failed to perform boolean match"), false);
```

Comment:
Nit: add error context from `e.what()` to help debugging

Reply:
Have made this change in the `append-header` branch, will go through and confirm that I didn't miss an error message once I merge the PRs up to that change.